src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -561,7 +561,8 @@
   [{:keys [server] :as webserver-context} :- ServerContext]
   (when-let [mbean-container (:mbean-container @(:state webserver-context))]
     (log/debug "Cleaning up JMX MBean container")
-    (.destroy mbean-container))
+    (.destroy mbean-container)
+    (swap! (:state webserver-context) assoc :mbean-container nil))
```

Comment:
Yeah, that seems like a reasonable suggestion, but I do think that we want to be in a world where it's considered a full-on bug in TK if lifecycle functions are happening on two threads at the same time, and fix that if it comes up.  So I'm inclined to think that it's not worth the extra precautions here?